sunpy/time/timerange.py

```diff
@@ -407,15 +407,24 @@ def extend(self, dt_start, dt_end):
 
         Parameters
         ----------
-        dt_start : `astropy.time.TimeDelta`
+        dt_start : `astropy.time.TimeDelta`, `datetime.timedelta`, `astropy.units.Quantity`
             The amount to shift the start time.
-        dt_end : `astropy.time.TimeDelta`
+        dt_end : `astropy.time.TimeDelta`, `datetime.timedelta`, `astropy.units.Quantity`
             The amount to shift the end time.
         """
-        # TODO: Support datetime.timedelta
+        if isinstance(dt_start, timedelta):
+            dt_start = TimeDelta(dt_start)
+            dt_start.format = self.dt.format
+        if isinstance(dt_end, timedelta):
+            dt_end = TimeDelta(dt_end)
+            dt_end.format = self.dt.format
```

Comment:
```suggestion
        dt_start = TimeDelta(dt_start, scale=self._t1.scale) if not isinstance(dt_start, TimeDelta) else dt_start
        dt_end = TimeDelta(dt_end, scale=self._t2.scale) if not isinstance(dt_end, TimeDelta) else dt_end
```
These shouldn't be a need to set the format, but there is a need to set the scale